### PR TITLE
Issue #126: Tried to fix Tuexen's issue with HE not checking that a c…

### PR DESCRIPTION
**Pull request by karlgrin**
Attempt to address issue #126 (Tüxen).

### CODE REVIEW

**Comment by oystedal** [comment]
Code looks good to me. @tuexen: Could you test this, please?

File: neat_core.c [hunk 1]
```diff
@@ -1170,6 +1170,7 @@ static void
 he_connected_cb(uv_poll_t *handle, int status, int events)
 {
     static unsigned int c = 0;
+
```
Comment (tuexen) [inline neat_core.c]: Why is this whitespace change needed?

File: neat_core.c [hunk 2]
```diff
@@ -1225,6 +1226,18 @@ he_connected_cb(uv_poll_t *handle, int status, int events)
...
+    if (getsockopt(candidate->pollable_socket->fd, SOL_SOCKET, SO_ERROR, &so_error, &len) < 0) {
+    
+        neat_log(NEAT_LOG_DEBUG, "Call to getsockopt failed: %s", strerror(errno));
+        neat_io_error(candidate->ctx, flow, NEAT_ERROR_INTERNAL);
+        return;
+    }
+    status = so_error;
+    neat_log(NEAT_LOG_DEBUG,
+             "Connection status: %d", status);
+
```
Comment (tuexen) [inline neat_core.c]: Looks good. I tested the change and it works.

**Comment by karlgrin** [comment]
Hi Tüxen!

As far as I know, completely unintentional.

BR
Karl-Johan Grinnemo
Lektor/Senior Lecturer
Datavetenskap/Computer Science
Karlstads universitet/Karlstad university
Tfn.:/Phone: +46 (0)54 700 24 40
Mobil:/Mobile: +46 (0)708 740 511

**Comment by tuexen** [comment]
So remove the line with the whitespace and merge the code.

Thanks for fixing the bug!

Best regards
Michael

**Pull request merged**